crates/bevy_pbr/src/environment_map/mod.rs

```diff
@@ -26,67 +58,451 @@ impl Plugin for EnvironmentMapPlugin {
 
         app.register_type::<EnvironmentMapLight>()
             .add_plugins(ExtractComponentPlugin::<EnvironmentMapLight>::default());
+
+        if let Ok(render_app) = app.get_sub_app_mut(RenderApp) {
+            render_app
+                .init_resource::<RenderEnvironmentMaps>()
+                .add_systems(
+                    Render,
+                    prepare_environment_maps.in_set(RenderSet::PrepareResources),
+                );
+        }
     }
 }
 
 /// Environment map based ambient lighting representing light from distant scenery.
 ///
-/// When added to a 3D camera, this component adds indirect light
-/// to every point of the scene (including inside, enclosed areas) based on
-/// an environment cubemap texture. This is similar to [`crate::AmbientLight`], but
-/// higher quality, and is intended for outdoor scenes.
+/// When added to a 3D camera, this component adds indirect light to every point of the scene based
+/// on an environment cubemap texture, if not overridden by a [LightProbe]. This is similar to
+/// [`crate::AmbientLight`], but higher quality, and is intended for outdoor scenes.
+///
+/// When added to a [LightProbe], the indirect light will be added to all meshes within the bounds
+/// of the light probe, overriding the camera's environment map if any.
 ///
 /// The environment map must be prefiltered into a diffuse and specular cubemap based on the
 /// [split-sum approximation](https://cdn2.unrealengine.com/Resources/files/2013SiggraphPresentationsNotes-26915738.pdf).
 ///
 /// To prefilter your environment map, you can use `KhronosGroup`'s [glTF-IBL-Sampler](https://github.com/KhronosGroup/glTF-IBL-Sampler).
+/// For a useful artist-friendly GUI, try pcwalton's [glTF IBL Sampler GUI](https://github.com/pcwalton/gltf-ibl-sampler-egui).
 /// The diffuse map uses the Lambertian distribution, and the specular map uses the GGX distribution.
-///
 /// `KhronosGroup` also has several prefiltered environment maps that can be found [here](https://github.com/KhronosGroup/glTF-Sample-Environments).
+///
+/// Note that all environment maps and reflection probes in the scene currently must have the same
+/// size, number of mipmap levels, and texture format. This is because they're packed into a
+/// texture array at runtime. A warning is emitted if this restriction isn't followed.
 #[derive(Component, Reflect, Clone, ExtractComponent)]
-#[extract_component_filter(With<Camera3d>)]
+#[extract_component_filter(Or<(With<Camera3d>, With<LightProbe>)>)]
 pub struct EnvironmentMapLight {
+    /// The blurry, often-low-resolution cubemap representing diffuse light.
     pub diffuse_map: Handle<Image>,
+    /// The sharp, high-resolution cubemap representing specular light.
```

Comment:
Eh that's misleading, the specular mips will also be blurry. Also applies to below.